### PR TITLE
Update for new Kaleidoscope plugin API.

### PR DESCRIPTION
**Pull request by bjc**

### CODE REVIEW

**Comment by cdisselkoen** [comment]
Hi, thanks very much for this update!  Have you tested this and confirmed it works with the latest Kaleidoscope?

File: src/Kaleidoscope/MacrosOnTheFly.cpp [hunk 1]
```diff
@@ -424,10 +418,10 @@ Key MacrosOnTheFly::eventHandlerHook(Key mapped_key, byte row, byte col, uint8_t
       play_col = col;
       currentState = PICKING_SLOT_FOR_PLAY;
     }
-    return Key_NoKey;
+    return kaleidoscope::EventHandlerResult::OK;
```
Comment (cdisselkoen) [inline src/Kaleidoscope/MacrosOnTheFly.cpp]: Everywhere else, `return Key_NoKey` was converted to `return kaleidoscope::EventHandlerResult::EVENT_CONSUMED`.  Was the difference here intentional?

Comment (bjc) [reply src/Kaleidoscope/MacrosOnTheFly.cpp]: Nope. I'll fix it up presently.

**Comment by bjc** [comment]
I'm using it right now, albeit not extensively. I've checked basic functionality (hit macro record key, letter, type stuff, end recording, then play it back), and the LEDs seem to work, although I'm not entirely sure how they're supposed to work, so can't say for sure.

I've had no issue with the actual recording and playing back of macros, though.

**Comment by cdisselkoen** [comment]
Sounds good.  I'll give this a little more testing and assuming all goes well, let's get this merged.

Another thing is that we should update the README to reflect the new plugin API as well.  (Particularly the "Adding the plugin to your sketch" section.)  I'm happy to do this separately after this is merged, but if you want to roll it in with this PR, you are more than welcome.

**Comment by cdisselkoen** [comment]
Sorry for the delay - works great!  Merging.

**Pull request merged**